build/lib/src/library_cycle_graph/phased_value.dart

```diff
@@ -142,16 +147,27 @@ abstract class PhasedValue<T>
 
 /// A [value] with optionally limited lifespan.
 ///
-/// If [expiresAt] is `null`, the value never expires.
+/// If [expiresAfter] is `null`, the value never expires.
 ///
-/// If [expiresAt] is set, the value expires at that phase: it takes a new value
-/// in the _next_ phase after that.
+/// If [expiresAfter] is set, the value expires after that phase, taking a new
+/// value in the next phase.
 abstract class ExpiringValue<T>
     implements Built<ExpiringValue<T>, ExpiringValueBuilder<T>> {
   T get value;
-  int? get expiresAt;
+  int? get expiresAfter;
 
-  factory ExpiringValue(T value, {int? expiresAt}) =>
-      _$ExpiringValue<T>._(value: value, expiresAt: expiresAt);
+  factory ExpiringValue(T value, {int? expiresAfter}) =>
+      _$ExpiringValue<T>._(value: value, expiresAfter: expiresAfter);
   ExpiringValue._();
 }
+
+/// Returns the earliest of two nullable phases [a] and [b].
+///
+/// `null` represents "never", so any non-`null` phase is earlier than a `null`
+/// one.
+int? earliestPhase(int? a, int? b) =>
+    a == null
+        ? b
+        : b == null
+        ? a
```

Comment:
Oh, now it's here instead.
Maybe it's just me, but having several nested (I forget what they're called -- Ternary conditional operator I think) just makes my eyes hurt.

```
int? earliestPhase(int? a, int? b) {
  if (a == null) return b;
  if (b == null) return a;
  return min(a, b);
}
```

is much more readable to me.